### ISSUE TITLE
bug: Wrong highlighting in some context

### ISSUE THREAD

**Issue by vdegenne** [folke/twilight.nvim#50]
### Did you check docs and existing issues?

- [X] I have read all the twilight.nvim docs
- [X] I have updated the plugin to the latest version before submitting this issue
- [X] I have searched the existing issues of twilight.nvim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.10.2

### Operating system/version

Fedora 41

### Describe the bug

![screenshot_2024-12-03_14-58-43](https://github.com/user-attachments/assets/f9cf51af-b34a-48ed-9ab6-5e0535e378c4)

I'm using a library that uses template literals for html code in JS, that's when it seems to break.

### Steps To Reproduce

1. Open a js file
2. Write some html in tagged templates, e.g. ``html`...` ``
3. In some parts the highlighting won't follow the open<->close context.

### Expected Behavior

Proper highlighting in html tagged literals.

**Comment by vdegenne** [comment]
Probably difficult to fix and maybe not a twilight issue if it's based on treesitter etc.  
 
But I have a suggestion to make (feature request). Having an option to change the highlight strategy (currently 'treesitter') and can be changed to 'indentscope' would be nice.
In the video, if you look the indentscope plugin vertical highlighted line, it always find the best region where the cursor is:


https://github.com/user-attachments/assets/a32942f9-eaa3-49b6-9ef5-b38cae592fd1

In other words, a strategy to highlight only parts sharing the same indentation level?

**Comment by vdegenne** [comment]
For reference, indentscope has an util function to retrieve scope information, this could be used to easily find the region to highlight: 

https://github.com/echasnovski/mini.indentscope/blob/da9af64649e114aa79480c238fd23f6524bc0903/lua/mini/indentscope.lua#L295

**Comment by folke** [comment]
If treesitter is not available it just uses prev/next lines as context.
Not a bug and doesn't need fixing.

**Comment by folke** [comment]
Is there any reason why you don't use Treesitter? It's superior in every way compared to vim synatx

**Comment by folke** [comment]
I should have mentioned that I'm working on a new version of twilight in snacks.

Actually a couple of new things in snacks coming up that are all related:
- scope: utility to determine the scope of the cursor based on indents and/or treesitter
- zen: zen mode
- dim: uses scope
- indent: indent guides and scope highlighting

**Comment by vdegenne** [comment]
Ok nice :) Thanks for your hard work
